15700093 SUNBT7023648 xscreensaver does not allow user to enter password
	 	      and login w/ sv_SE.ISO8859-01/-15

Not applicable upstream - specific to our added GTK unlock dialog.
---
 driver/lock-Gtk.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/driver/lock-Gtk.c b/driver/lock-Gtk.c
--- a/driver/lock-Gtk.c
+++ b/driver/lock-Gtk.c
@@ -659,6 +659,7 @@ main (int argc, char *argv[])
   GConfClient *client;
   const char *modulesptr = NULL;
   int i;
+  const char *locale = NULL;
 
   gboolean  at_enable  = FALSE; /* accessibility mode enabled ? */
   Bonobo_ServerInfoList *server_list = NULL;
@@ -949,6 +950,8 @@ main (int argc, char *argv[])
   gtk_widget_grab_focus (pwd->user_input_entry);
 
   ioc = g_io_channel_unix_new (0);
+  g_get_charset (&locale);
+  g_io_channel_set_encoding(ioc, locale, NULL);
   g_io_add_watch (ioc, G_IO_IN | G_IO_HUP, handle_input, pwd);
 
   gtk_main ();

